File: Data Structures & Algorithms/validate-parentheses/submission-7.cpp
class Solution {
public:
    bool isValid(string& s) {
        stack<char> x{};

        for (auto& c : s) {
            if ((c == '(') ||
                (c == '{') ||
                (c == '[')) {
                x.push(c);
            } else {
                if (x.empty()) {
                    return false;
                }

                auto& t = x.top();
                x.pop();

                if ((c == ')' && t != '(') ||
                    (c == '}' && t != '{') ||
                    (c == ']' && t != '[')) {
                    return false;
                }
            }
        }

        return x.empty();
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-7.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> values = new();

        int first = 0;
        int second = 0;

        foreach(string token in tokens) {
            if (token == "+" || token == "-" || token == "*" || token == "/") {
                second = values.Pop();
                first = values.Pop();

                if (token == "+") {
                    values.Push(first + second);
                } 
                else if (token == "-") {
                    values.Push(first - second);
                }
                else if (token == "*") {
                    values.Push(first * second);
                } 
                else {
                    values.Push(first / second);
                }
            } else {
                values.Push(int.Parse(token));
            }
        }

        return values.Pop();
    }
}
